frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/types.ts

```diff
@@ -17,6 +17,49 @@ export type PipelineRunAfterNodeModelData = {
     runAfter?: string[];
   };
 };
+
+type FinallyTask = {
+  name: string;
+  runAfter?: string[];
+  error?: string;
+  selected?: boolean;
+  disableTooltip?: boolean;
+  onTaskSelection?: () => void;
+};
+type FinallyListTask = {
+  name: string;
+  convertList: (resource: TaskKind) => void;
+  onRemoveTask: () => void;
+};
+type FinallyNodeTask = {
+  name: string;
+  runAfter: string[];
+  selected?: boolean;
+  isFinallyTask: boolean;
+  finallyTasks: FinallyTask[];
+};
+export type FinallyNodeData = {
+  task: FinallyNodeTask;
+};
+export type FinallyBuilderNodeData = {
```

Comment:
Consistency would be helpful for future lookup of files/types. `Finally + Builder` or `Builder + Finally`. Your component is `BuilderFinallyNode`, your type is `FinallyBuilderNode`